source/functions/BEMIO/checkAndFormatPlotVars.m

```diff
@@ -0,0 +1,71 @@
+function varargout = checkAndFormatPlotVars(varargin)
+% Checks and formats the BEM plotting variables (plotBodies, plotDofs, 
+% plotDirections) to ensure desired plots line up with existing BEM data 
+% and format properly for plotting functions.
+%
+% This function is not called directly by the user; it is automatically
+% implemented within the BEM plotting functions.
+%
+% Parameters
+% ----------
+%     varargin : struct(s)
+%         The hydroData structure(s) created by the other BEMIO functions.
+%         One or more may be input.
+%
+% Returns
+% -------
+%     varargout : struct(s)
+%         The hydroData structure(s) with checked and formatted plotting
+%         variables. The number of output structures is equal to the number 
+%         of input structures.
+% 
+
+numHydro = length(varargin{1});
+
+for ii = 1:numHydro
+    % Format plotting dofs
+    if isequal(size(varargin{1}{ii}.plotDofs),[1 2]) && range(varargin{1}{ii}.plotDofs) == 0
```

Comment:
``range`` requires other toolboxes. Also this line technically doesn't do anything, so let's remove it

Reply:
Didn't realize it required toolboxes, thanks! I added this line to prevent multiple of the same plots from being plotted. For example if you do plotDofs = [3,3] or [3;3], it'll just plot 2 of the same plot without this line. Not a big deal at all, but thought it was a simple fix so I'll keep it in (without using range) unless there is another reason to remove it